File: dataset/avatar/Python/Code/atcoder_ABC178_B.py
import math , itertools , fractions , heapq , collections , bisect , sys , queue , copy
sys.setrecursionlimit ( 10 ** 7 )
inf = 10 ** 20
mod = 10 ** 9 + 7
dd = [ ( - 1 , 0 ) , ( 0 , 1 ) , ( 1 , 0 ) , ( 0 , - 1 ) ]
ddn = [ ( - 1 , 0 ) , ( - 1 , 1 ) , ( 0 , 1 ) , ( 1 , 1 ) , ( 1 , 0 ) , ( 1 , - 1 ) , ( 0 , - 1 ) , ( - 1 , - 1 ) ]
def LI ( ) : return [ int ( x ) for x in sys.stdin.readline ( ).split ( ) ]
def I ( ) : return int ( sys.stdin.readline ( ) )
def F ( ) : return float ( sys.stdin.readline ( ) )
def LS ( ) : return sys.stdin.readline ( ).split ( )
def S ( ) : return input ( )
def main ( ) :
    a , b , c , d = LI ( )
    return max ( a * c , b * d , a * d , b * c )
print ( main ( ) )

File: output/gpt_4o/avatar/direct_translation/python/java/atcoder_ABC178_B.java

import java.util.*;
import java.io.*;

public class atcoder_ABC178_B {
    static int inf = (int) Math.pow(10, 20);
    static int mod = (int) Math.pow(10, 9) + 7;
    static int[][] dd = {{-1, 0}, {0, 1}, {1, 0}, {0, -1}};
    static int[][] ddn = {{-1, 0}, {-1, 1}, {0, 1}, {1, 1}, {1, 0}, {1, -1}, {0, -1}, {-1, -1}};

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int a = sc.nextInt();
        int b = sc.nextInt();
        int c = sc.nextInt();
        int d = sc.nextInt();
        System.out.println(Math.max(Math.max(a * c, b * d), Math.max(a * d, b * c)));
    }
}

